Content.Shared/Doors/Components/FirelockComponent.cs

```diff
@@ -7,9 +8,15 @@ namespace Content.Shared.Doors.Components
     /// auto-closing on depressurization, air/fire alarm interactions, and preventing normal door functions when
     /// retaining pressure..
     /// </summary>
-    [RegisterComponent]
+    [RegisterComponent, NetworkedComponent, AutoGenerateComponentState]
     public sealed partial class FirelockComponent : Component
     {
+        /// <summary>
+        /// Networked powered status for firelock to allow for proper client prediction
+        /// </summary>
+        [DataField, AutoNetworkedField]
+        public bool Powered;
+
         /// <summary>
```

Comment:
I don't really like adding even more Powered fields to components as these will just need ripping out later, it should really be on some separate shared component so multiple systems can take advantage of it.

Reply:
I'm not sure how I feel about this. Are we really making things simpler by doing this? We're taking a boolean flag to be replicated and changing it into a full fat component. All we're doing right now is listening for an event and setting a flag. Now we'd be listening for an event and adding a component. That seems like we're doing more work to achieve the same thing. Maybe it allows us to merge the APIs...but I'm skeptical to that. I suppose it would be some sort of powered system that listens for a Power change event and adds a powered component to an object? Possibly...I would be interested to know if that would cover all existing cases, and if it's saving that much of a headache. 

Regardless, I'm still new to this codebase so if it's believed to be the best course of action I'm happy to make the change of course! Just wary of over engineering something.

Reply:
Although, thinking about the idea for debug that powering something could be as easy as adding a component...that actually sounds pretty useful. So now I'm talking myself into it a bit more :laughing: